src/Field/Definition/RichTextField.php

```diff
@@ -108,18 +108,31 @@ public function transformData (
 	{
 		\assert(null === $data || \is_array($data));
 
-		$transformed = $data;
+		$transformed = null !== $data && !$this->contentIsEmpty($data)
+			? $data
+			: null;
 
-		if (null !== $data
-			&& 1 === \count($data["content"])
-			// fetch the first (and only) content and check if it is empty
-			&& null === $context->dataTransformer->normalizeOptionalString($data["content"][0]["content"][0]["text"] ?? null)
-		)
+		$dataVisitor?->onDataVisit($this, $transformed);
+		return $transformed;
+	}
+
+	/**
+	 * Checks whether the given content is empty
+	 */
+	private function contentIsEmpty (array $data) : bool
+	{
+		$paragraphs = $data["content"];
+
+		if (\count($paragraphs) > 1)
 		{
-			$transformed = null;
+			return false;
 		}
 
-		$dataVisitor?->onDataVisit($this, $transformed);
-		return $transformed;
+		$firstItem = $data["content"][0] ?? [];
+		$firstItemType = $firstItem["type"] ?? null;
+		$firstItemContent = $firstItem["content"] ?? [];
+
+		return 0 === \count($firstItemContent) &&
+			\in_array($firstItemType, ["paragraph", "heading"], true);
```

Comment:
Isn't this potentially prone to false-positives? When an editor clicks on the „list items” formatting option but doesn't enter anything, this might fail, no?

Reply:
It can't be truly empty:

![CleanShot 2023-04-12 at 10 31 11](https://user-images.githubusercontent.com/1032411/231399993-d741a539-c3c8-4315-a594-b0d896152dae.png)

So you see that it's not empty.
If you remove anything more (and it's visually empty), it can't be anything except for paragraph / heading - at least according to my tests.